### DISCUSSION TITLE
EFCore in .Net 8 on iOS and NativeAOT

### DISCUSSION THREAD

**Discussion by danbrannanavontus** [dotnet/maui#21909] [Upgrading from Xamarin]
Hi

We're porting an application from Xamarin/.Net4.x to Maui/.Net 8 and after months of wrestling with various iOS issues, the app now builds, installs and runs. However, upon performing any data access, via Microsoft.EntityFrameworkCore 8.0.3, we now get the following error: 'Model building is not supported when publishing with NativeAOT. Use a compiled model'. Turning off AOT using &#x3c;PublishAot&#x3e;false&#x3c;/PublishAot&#x3e; doesn't seem to do anything.

From what I have read, NativeAOT is an experimental new feature in .Net 7/8, and that proper support for it in EFCore will come from .Net 9 onwards. I also read that AOT cannot be disabled in iOS which does seem to be true given disabling &#x3c;PublishAot&#x3e; doesn't fix it. So does this mean that EFCore simply doesn't work in .Net 8 on iOS? How did EF ever work in Xamarin if AOT can't be disabled?

I hope someone can point me in the right direction on this one. 

Thanks!

Dan

**Comment by danbrannanavontus** [comment]
Quick update. Downgrading to EFCore 7 gives a similar but different error: 'Attempting to JIT compile method wrapper dynamic-method while running in aot-only mode'

**Reply by bluedot2** [reply]
FYI, I am successfully using EFCore 7 with MAUI/.NET 7 (I think NativeAot support for iOS only came with 8.0). You might want to use that as an interim solution.

However, I would still like to see an answer to your original question from the Microsoft folks, as I am also confused as to why we can't turn NativeAOT off in .NET 8.

**Comment by jmichas** [comment]
For anyone who finds this, I used the fix described in this report and it worked for me.
https://github.com/dotnet/maui/issues/23653
